feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/thread/MetisThreadUi.kt

```diff
@@ -110,9 +112,21 @@ internal fun MetisThreadUi(
                     else -> throw NotImplementedError()
                 }
             },
+            onResolvePost = { post ->
+                val parentPost = postDataState.orNull()
+
+                if (post is AnswerPostPojo) {
```

Comment:
Isn't IAnswerPost sufficient for this check?

Reply:
I don't think it's needed, would you prefer this instead?
```
onResolvePost = { post ->
    val parentPost = postDataState.orNull()
                    ?: return@MetisReplyHandler CompletableDeferred(
                        MetisModificationFailure.UPDATE_POST)

     viewModel.toggleResolvePost(parentPost, post as AnswerPostPojo)
},
```

Reply:
`post as AnswerPostPojo` this line does not feel safe though.

Reply:
I think you should redesign your toggleResolvePost function in the viewmodel to not take AnswerPostPojo as a parameter. Instead, I suggest the function should only take the server id of the answer post as a parameter. The viewmodel function can then resolve the data needed for the request using the server id.

Reply:
As a rule of thumb, as soon as you start doing unsafe casting your code design is wrong.